crates/bin/c8y_configuration_poc/src/c8y_http_proxy/actor.rs

```diff
@@ -37,25 +37,27 @@ use tedge_http_ext::HttpResult;
 
 const RETRY_TIMEOUT_SECS: u64 = 60;
 
-pub(crate) struct C8YHttpProxyActor {
+struct C8YHttpProxyActor {
     end_point: C8yEndPoint,
     child_devices: HashMap<String, String>,
+    peers: C8YHttpProxyMessageBox,
 }
 
 #[async_trait]
-impl Actor for C8YHttpProxyActor {
+impl Actor for C8YHttpConfig {
```

Comment:
This might be controversial!

The goal is to test the idea of having an actor struct owning both the actor state (e.g the `end_point` definition and the map of  `child_devices`) and the message box (to avoid passing this box to all and every method calls).

So one pretends that `C8YHttpConfig` is the `Actor` while the point is only to defer the real actor construction till the actor is actually spawned.

Reply:
That looks really weird. When the `MessageBox` is part of the state, why pass it in the `run` method signature as well separately? And if that's really necessary, why not pass an `&mut Self::Messagebox` as the argument instead of this hack?

Reply:
> When the MessageBox is part of the state, why pass it in the run method signature as well separately?

The point here is to see if this simplify the actor implementation (it does), without changing the `Actor` trait only for  that exploratory test.

> In that case we can still build the actor "naturally", by passing both the config and the message box in the ActorBuilder::spawn.

This is what is done.

Reply:
As a follow-up we will give a try to the idea to have the message box owned by the actor rather than passed as an argument to the run method.